app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt

```diff
@@ -163,15 +171,75 @@ class NewMessageFragment : Fragment() {
     }
 
     private fun initDraftAndUi() {
-        newMessageViewModel.initDraftAndUi(newMessageActivityArgs)
-            .observe(viewLifecycleOwner) { isSuccess ->
-                if (isSuccess) {
-                    observeContacts()
-                    populateUiWithExistingDraftData()
-                } else {
-                    requireActivity().finish()
-                }
+        newMessageViewModel.initDraftAndUi(newMessageActivityArgs).observe(viewLifecycleOwner) { isSuccess ->
+            if (isSuccess) {
+                observeContacts()
+                populateUiWithExistingDraftData()
+                handleActionSend()
+            } else {
+                requireActivity().finish()
+            }
+        }
+    }
+
+    private fun handleActionSend() {
```

Comment:
Why don't you do this in the activity directly?

Reply:
Its clean to have all the shares in one place, a reason for the activity?
Otherwise the code will look like magic 🤔